third_party/terraform/resources/resource_compute_instance.go.erb

```diff
@@ -1387,14 +1414,17 @@ func resourceComputeInstanceUpdate(d *schema.ResourceData, meta interface{}) err
 			}
 		}
 
-		if d.HasChange(prefix + ".access_config") {
-
-			// TODO: This code deletes then recreates accessConfigs.  This is bad because it may
-			// leave the machine inaccessible from either ip if the creation part fails (network
-			// timeout etc).  However right now there is a GCE limit of 1 accessConfig so it is
-			// the only way to do it.  In future this should be revised to only change what is
-			// necessary, and also add before removing.
+		readFingerPrint := func() error {
+			// re-read fingerprint
+			instance, err = config.clientComputeBeta.Instances.Get(project, zone, instance.Name).Do()
+			if err != nil {
+				return err
+			}
+			instNetworkInterface = instance.NetworkInterfaces[i]
```

Comment:
if you were at the call site of this function, would you realize it's going to update a variable that's declared outside the function? I think having side-effects like this is probably not the best road to go down, and it's better just to be explicit about inputs and outputs.

Reply:
changed implementation to encapsulate the state